Tests/ASCollectionViewTests.mm

```diff
@@ -1071,19 +1071,15 @@ - (void)testInitialRangeBoundsWithCellLayoutMode:(ASCellLayoutMode)cellLayoutMod
   // Trigger the initial reload to start 
   [window layoutIfNeeded];
 
-  // Test the APIs that monitor ASCollectionNode update handling if collection node should
-  // layout asynchronously
-  if (![cn.view dataController:nil shouldSynchronouslyProcessChangeSet:nil]) {
```

Comment:
@maicki Heads up that I think we need to remove this line.

I understand why we need to call `waitUntilAllUpdatesAreProcessed` if the collection view doesn't synchronously process the change set: to make sure the initial load finishes before we can proceed. However, in order to reliably do so, we need to pass in a valid change set that contains a `reloadData` change, instead of `nil`.

We can either do that, or always call `waitUntilAllUpdatesAreProcessed` like what I'm proposing here.
- If the change set is processed asynchronously, calling `waitUntilAllUpdatesAreProcessed` will be required here.
- If it's processed synchronously, calling `waitUntilAllUpdatesAreProcessed` here will be super fast and won't do any harm.